### PR TITLE
FIX: Ensure that our restorer is backwards compatible.

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

**Comment by tgxworld** [comment]
cc @ZogStriP Please review :grin:  Thanks!

**Comment by coding-horror** [comment]
Yes please, as this is blocking users as well.

**Comment by tgxworld** [comment]
I'll ship the fix with the new backup and restore format

File: lib/backup_restore/restorer.rb [hunk 1]
```diff
@@ -106,10 +106,19 @@ def initialize_state
...
       @tmp_directory = File.join(Rails.root, "tmp", "restores", @current_db, @timestamp)
+      @source_filename = File.join(Backup.base_directory, @filename)
       @archive_filename = File.join(@tmp_directory, @filename)
       @tar_filename = @archive_filename[0...-3]
       @meta_filename = File.join(@tmp_directory, BackupRestore::METADATA_FILE)
-      @dump_filename = File.join(@tmp_directory, BackupRestore::DUMP_FILE)
+
+      # For backwards compatibility
+      @dump_filename =
+        if system("tar --list --verbose --file #{@source_filename} #{BackupRestore::DUMP_FILE}")
```
Comment (ZogStriP) [inline lib/backup_restore/restorer.rb]: Do we need to do a `--verbose` when we only care about the exit status of the command?

Comment (tgxworld) [inline lib/backup_restore/restorer.rb]: Nope hehe. I'll remove it tml ")

**Comment by ZogStriP** [comment]
LGTM

**Pull request merged**